src/core/qgssettings.h

```diff
@@ -0,0 +1,197 @@
+     * In addition, query functions such as childGroups(), childKeys(), and allKeys()
+     * are based on the group. By default, no group is set.
+     */
+    void beginGroup( const QString &prefix );
+    //! Resets the group to what it was before the corresponding beginGroup() call.
+    void endGroup();
+    //! Returns a list of all keys, including subkeys, that can be read using the QSettings object.
+    QStringList allKeys() const;
+    //! Returns a list of all top-level keys that can be read using the QSettings object.
+    QStringList childKeys() const;
+    //! Returns a list of all key top-level groups that contain keys that can be read using the QSettings object.
+    QStringList childGroups() const;
+    //! Return the path to the Global Settings QSettings storage file
+    static QString globalSettingsPath() { return sGlobalSettingsPath; }
+    //! Set the Global Settings QSettings storage file
+    static bool setGlobalSettingsPath( QString path );
+    //! Adds prefix to the current group and starts reading from an array. Returns the size of the array.
+    int beginReadArray( const QString &prefix );
+    //! Closes the array that was started using beginReadArray() or beginWriteArray().
+    void endArray();
+    //! Sets the current array index to i. Calls to functions such as setValue(), value(), remove(), and contains() will operate on the array entry at that index.
+    void setArrayIndex( int i );
+    //! Sets the value of setting key to value. If the key already exists, the previous value is overwritten.
+    //! @note keys are case insensitive
+    void setValue( const QString &key, const QVariant &value );
+
+    /** Returns the value for setting key. If the setting doesn't exist, it will be
+     * searched in the Global Settings and if not found, returns defaultValue.
+     *If no default value is specified, a default QVariant is returned.
+     */
+    QVariant value( const QString &key, const QVariant &defaultValue = QVariant() ) const;
+    //! Returns true if there exists a setting called key; returns false otherwise.
+    //! If a group is set using beginGroup(), key is taken to be relative to that group.
+    bool contains( const QString &key ) const;
+    //! Returns the path where settings written using this QSettings object are stored.
+    QString fileName() const;
+    //! Writes any unsaved changes to permanent storage, and reloads any settings that have been
+    //! changed in the meantime by another application.
+    //! This function is called automatically from QSettings's destructor and by the event
+    //! loop at regular intervals, so you normally don't need to call it yourself.
+    void sync();
+    //! Removes the setting key and any sub-settings of key.
+    void remove( const QString &key );
+    //! Overloaded getter that accepts an additional Section argument
+    QVariant sectionValue( const QString &key, const Section section, const QVariant &defaultValue = QVariant() ) const;
```

Comment:
Thanks for the tip, I'll add the `Section` as an optional last argument to value getter/setter.